tests/system_tests_three_routers.py

```diff
@@ -373,5 +433,189 @@ def run(self):
+        self.route_check_address    = route_check_address
+        self.true_statement         = "This IS the message you are looking for."
+        self.error                  = None
+        self.send_connection        = None
+        self.recv_connection        = None
+        self.route_check_connection = None
+        self.route_check_sender     = None
+        self.sender                 = None
+        self.route_check_timer      = None
+        self.route_check_receiver   = None
+        self.count                  = 1
+        self.sent                   = 0
+        self.route_checks_sent      = 0
+
+
+    def bail ( self, error_text ) :
+        self.error = error_text
+        self.recv_connection.close()
+        self.route_check_connection.close()
+        self.send_connection.close()
+        self.timer.cancel()
+        if self.route_check_timer :
+            self.route_check_timer.cancel()
+            self.route_check_timer = None
+
+
+    def timeout(self):
+        self.bail ( "Timeout Expired" )
+
+
+    def poll_timeout ( self ) :
+        self.poll()
+
+
+    def on_start(self, event):
+        # Iff this timer expires, the test fails.
+        self.timer = event.reactor.schedule ( TIMEOUT, Timeout(self) )
+
+        # At startup, create only the receivers and the sender that checks for
+        # the route being ready.  Creation of the payload message sender
+        # has to wait until we know the address is available, to avoid a
+        # 'no route to destination' error.
+        self.recv_connection = event.container.connect ( self.route_container )
+
+        self.route_check_connection  = event.container.connect ( self.route_check_address )
+        self.route_check_receiver = event.container.create_receiver ( self.route_check_connection,
+                                                                dynamic=True
+                                                              )
+        self.route_check_sender = event.container.create_sender ( self.route_check_connection,
+                                                                  "$management"
+                                                                )
+
+
+
+    def poll ( self ):
+        # The router prepends this 'D' to the address.  If we want
+        # to look for it down at router A, we have to do likewise.
+        doctored_linkroute_prefix = 'D' + self.linkroute_prefix
+        request = self.proxy.read_address ( doctored_linkroute_prefix )
+        self.route_check_sender.send ( request )
+
+
+
+    def on_link_opened ( self, event ):
+
+        if event.receiver:
+            event.receiver.flow ( self.count )
+
+        if event.receiver == self.route_check_receiver:
+            self.proxy = RouterProxy ( self.route_check_receiver.remote_source.address )
+            self.poll()
+
+
+    def on_link_opening ( self, event ):
+        if event.receiver:
+            event.receiver.target.address = self.route_container
```

Comment:
self.route_container is a host address (0.0.0.0:<some-port>), not an AMQP address.  If you look at the trace of the test, you will see that the addresses in the paired attach frames are different.  This should not be the case.